Example/ViewController.swift

```diff
@@ -450,12 +451,13 @@ class ViewController: UIViewController {
         return navigationViewController
     }
     
-    func present(_ navigationViewController: NavigationViewController, completion: CompletionHandler? = nil) {
+    func presentAndRemoveMapview(_ navigationViewController: NavigationViewController, completion: CompletionHandler? = nil) {
         navigationViewController.modalPresentationStyle = .fullScreen
         activeNavigationViewController = navigationViewController
         
         present(navigationViewController, animated: true) {
             completion?()
+            self.navigationMapView = nil
```

Comment:
We only want to remove the map view from the view hierarchy, not from memory. Call `uninstall(_:)` instead of nilling out this property. After all, once the trip ends, the application goes back to this view controller, which needs to show the same map view again.

Reply:
But if we only remove the map view from the view hierarchy without set it to `nil`, the default `locationProvider` overridden by `PassiveLocationManager` will continue sending location updates, which leads to rerouting.

Reply:
Is there no way to stop a location provider while the map is in memory?

Reply:
Looks like this is what we did in v1._x_, so I suppose it’s fine to deallocate the map view then:

https://github.com/mapbox/mapbox-navigation-ios/blob/41525d9487fbd26e6b3cb3817aa1112ec69e73d4/Example/ViewController.swift#L377

Reply:
Yes. The `navigationMapView ` has a property observer to remove itself from its superview by calling the `func uninstall()` . So it would stop the location updating and reduce the memory usage at the same time.

https://github.com/mapbox/mapbox-navigation-ios/blob/92823986228eb1ec7ee8413562b18298d68cb777/Example/ViewController.swift#L32-L36